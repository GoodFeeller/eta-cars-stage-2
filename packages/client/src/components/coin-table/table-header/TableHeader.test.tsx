import { FunctionComponent, useState } from 'react'
import { sortType } from '../../../hooks/useSortCoins'
import TableHeader from './TableHeader'
import { render, screen, act } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

const TestComponent: FunctionComponent<{ initialState: sortType }> = ({
  initialState,
}) => {
  const [sort, setSort] = useState<sortType>(initialState)
  return (
    <>
      <article data-testid="property">{sort.property}</article>
      <article data-testid="to">{sort.to}</article>
      <table>
        <TableHeader sort={sort} setSort={setSort} />
      </table>
    </>
  )
}
describe('TableHeader test', () => {
  test('Rank test up', async () => {
    await act(async () =>
      render(
        <TestComponent initialState={{ property: 'priceUsd', to: 'down' }} />,
      ),
    )
    const prop = screen.queryByText('#') as HTMLElement
    userEvent.click(prop)
    const property = screen.queryByTestId('property') as HTMLElement
    const to = screen.queryByTestId('to') as HTMLElement
    const head = screen.queryAllByTestId('thead')
    expect(property.textContent).toContain('rank')
    expect(to.textContent).toContain('up')
    expect(head).toMatchSnapshot()
  })
  test('Rank test down', async () => {
    await act(async () =>
      render(
        <TestComponent initialState={{ property: 'priceUsd', to: 'down' }} />,
      ),
    )
    const prop = screen.queryByText('#') as HTMLElement
    userEvent.click(prop)
    userEvent.click(prop)
    const property = screen.queryByTestId('property') as HTMLElement
    const to = screen.queryByTestId('to') as HTMLElement
    const head = screen.queryAllByTestId('thead')
    expect(property.textContent).toContain('rank')
    expect(to.textContent).toContain('down')
    expect(head).toMatchSnapshot()
  })
  test('PriceUsd test up', async () => {
    await act(async () =>
      render(<TestComponent initialState={{ property: 'rank', to: 'down' }} />),
    )
    const prop = screen.queryByText('Price') as HTMLElement
    userEvent.click(prop)
    const property = screen.queryByTestId('property') as HTMLElement
    const to = screen.queryByTestId('to') as HTMLElement
    const head = screen.queryAllByTestId('thead')
    expect(property.textContent).toContain('priceUsd')
    expect(to.textContent).toContain('up')
    expect(head).toMatchSnapshot()
    userEvent.click(prop)
    userEvent.click(prop)
    expect(to.textContent).toContain('up')
  })
  test('PriceUsd test down', async () => {
    await act(async () =>
      render(<TestComponent initialState={{ property: 'rank', to: 'down' }} />),
    )
    const prop = screen.queryByText('Price') as HTMLElement
    userEvent.click(prop)
    userEvent.click(prop)
    const property = screen.queryByTestId('property') as HTMLElement
    const to = screen.queryByTestId('to') as HTMLElement
    const head = screen.queryAllByTestId('thead')
    expect(property.textContent).toContain('priceUsd')
    expect(to.textContent).toContain('down')
    expect(head).toMatchSnapshot()
  })
  test('marketCapUsd test up', async () => {
    await act(async () =>
      render(<TestComponent initialState={{ property: 'rank', to: 'down' }} />),
    )
    const prop = screen.queryByText('Market Cap') as HTMLElement
    userEvent.click(prop)
    const property = screen.queryByTestId('property') as HTMLElement
    const to = screen.queryByTestId('to') as HTMLElement
    const head = screen.queryAllByTestId('thead')
    expect(property.textContent).toContain('marketCapUsd')
    expect(to.textContent).toContain('up')
    expect(head).toMatchSnapshot()
  })
  test('marketCapUsd test down', async () => {
    await act(async () =>
      render(<TestComponent initialState={{ property: 'rank', to: 'down' }} />),
    )
    const prop = screen.queryByText('Market Cap') as HTMLElement
    userEvent.click(prop)
    userEvent.click(prop)
    const property = screen.queryByTestId('property') as HTMLElement
    const to = screen.queryByTestId('to') as HTMLElement
    const head = screen.queryAllByTestId('thead')
    expect(property.textContent).toContain('marketCapUsd')
    expect(to.textContent).toContain('down')
    expect(head).toMatchSnapshot()
  })
  test('changePercent24Hr test up', async () => {
    await act(async () =>
      render(<TestComponent initialState={{ property: 'rank', to: 'down' }} />),
    )
    const prop = screen.queryByText('24h%') as HTMLElement
    userEvent.click(prop)
    const property = screen.queryByTestId('property') as HTMLElement
    const to = screen.queryByTestId('to') as HTMLElement
    const head = screen.queryAllByTestId('thead')
    expect(property.textContent).toContain('changePercent24Hr')
    expect(to.textContent).toContain('up')
    expect(head).toMatchSnapshot()
  })
  test('changePercent24Hr test down', async () => {
    await act(async () =>
      render(<TestComponent initialState={{ property: 'rank', to: 'down' }} />),
    )
    const prop = screen.queryByText('24h%') as HTMLElement
    userEvent.click(prop)
    userEvent.click(prop)
    const property = screen.queryByTestId('property') as HTMLElement
    const to = screen.queryByTestId('to') as HTMLElement
    const head = screen.queryAllByTestId('thead')
    expect(property.textContent).toContain('changePercent24Hr')
    expect(to.textContent).toContain('down')
    expect(head).toMatchSnapshot()
  })
})
