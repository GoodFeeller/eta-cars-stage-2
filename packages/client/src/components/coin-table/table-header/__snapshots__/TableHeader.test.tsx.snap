// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeader test PriceUsd test down 1`] = `
Array [
  <thead
    data-testid="thead"
  >
    <tr>
      <th>
        #
      </th>
      <th
        class="imgTd"
      >
        Name
      </th>
      <th>
        Price (down)
      </th>
      <th>
        Market Cap
      </th>
      <th>
        24h%
      </th>
      <th />
    </tr>
  </thead>,
]
`;

exports[`TableHeader test PriceUsd test up 1`] = `
Array [
  <thead
    data-testid="thead"
  >
    <tr>
      <th>
        #
      </th>
      <th
        class="imgTd"
      >
        Name
      </th>
      <th>
        Price (up)
      </th>
      <th>
        Market Cap
      </th>
      <th>
        24h%
      </th>
      <th />
    </tr>
  </thead>,
]
`;

exports[`TableHeader test Rank test down 1`] = `
Array [
  <thead
    data-testid="thead"
  >
    <tr>
      <th>
        # (down)
      </th>
      <th
        class="imgTd"
      >
        Name
      </th>
      <th>
        Price
      </th>
      <th>
        Market Cap
      </th>
      <th>
        24h%
      </th>
      <th />
    </tr>
  </thead>,
]
`;

exports[`TableHeader test Rank test up 1`] = `
Array [
  <thead
    data-testid="thead"
  >
    <tr>
      <th>
        # (up)
      </th>
      <th
        class="imgTd"
      >
        Name
      </th>
      <th>
        Price
      </th>
      <th>
        Market Cap
      </th>
      <th>
        24h%
      </th>
      <th />
    </tr>
  </thead>,
]
`;

exports[`TableHeader test changePercent24Hr test down 1`] = `
Array [
  <thead
    data-testid="thead"
  >
    <tr>
      <th>
        #
      </th>
      <th
        class="imgTd"
      >
        Name
      </th>
      <th>
        Price
      </th>
      <th>
        Market Cap
      </th>
      <th>
        24h% (down)
      </th>
      <th />
    </tr>
  </thead>,
]
`;

exports[`TableHeader test changePercent24Hr test up 1`] = `
Array [
  <thead
    data-testid="thead"
  >
    <tr>
      <th>
        #
      </th>
      <th
        class="imgTd"
      >
        Name
      </th>
      <th>
        Price
      </th>
      <th>
        Market Cap
      </th>
      <th>
        24h% (up)
      </th>
      <th />
    </tr>
  </thead>,
]
`;

exports[`TableHeader test marketCapUsd test down 1`] = `
Array [
  <thead
    data-testid="thead"
  >
    <tr>
      <th>
        #
      </th>
      <th
        class="imgTd"
      >
        Name
      </th>
      <th>
        Price
      </th>
      <th>
        Market Cap (down)
      </th>
      <th>
        24h%
      </th>
      <th />
    </tr>
  </thead>,
]
`;

exports[`TableHeader test marketCapUsd test up 1`] = `
Array [
  <thead
    data-testid="thead"
  >
    <tr>
      <th>
        #
      </th>
      <th
        class="imgTd"
      >
        Name
      </th>
      <th>
        Price
      </th>
      <th>
        Market Cap (up)
      </th>
      <th>
        24h%
      </th>
      <th />
    </tr>
  </thead>,
]
`;
